libcxx/test/benchmarks/algorithms/lexicographical_compare.bench.cpp

```diff
@@ -19,7 +21,8 @@ static void bm_lexicographical_compare(benchmark::State& state) {
   for (auto _ : state) {
     benchmark::DoNotOptimize(vec1);
     benchmark::DoNotOptimize(vec2);
-    benchmark::DoNotOptimize(std::lexicographical_compare(vec1.begin(), vec1.end(), vec2.begin(), vec2.end()));
+    bool result = std::lexicographical_compare(vec1.begin(), vec1.end(), vec2.begin(), vec2.end());
+    benchmark::DoNotOptimize(result);
```

Comment:
What is the problem here?

Reply:
`DoNotOptimize` shouldn't be called with a temporary. A warning is emitted, but before running this stuff as part of the test suite, we just ignore that warning. After, it gets turned into an error.

Reply:
@ldionne The problem isn't the temporary. It's `const` values being passed, at least according to https://github.com/google/benchmark/pull/1493. This has been fixed by https://github.com/google/benchmark/pull/1608 in v1.8.1. Are you possibly using v1.8.0?

Reply:
The version of GoogleBenchmarks used here is the one in the monorepo under `third-party/benchmark`. It looks like that is 1.8.3 right now based on `third-party/benchmark/MODULE.bazel`.

Reply:
Removing this diff seems not to make any difference. I'll remove it.

Reply:
I suspect what happened is that I started this patch before GoogleBenchmark was updated to 1.8.3 in the monorepo. I went through the patch and removed the changes that weren't necessary anymore related to temporaries.